File: reverse.php
$numbers = range(1, 9);
shuffle($numbers);
$sorted = $numbers;
sort($sorted);
$steps = 0;

while ($numbers != $sorted)
    {
    print implode(" ", $numbers) . "\n";
    print "Reverse how many? ";
    $flipcount = (int)trim(fgets(STDIN));
    array_splice($numbers, 0, $flipcount, array_reverse(array_slice($numbers, 0, $flipcount)));
    $steps++;
    }

print "Done! That took you $steps steps.\n";

File: reverse.py
import random

numbers = random.sample(range(1,10), 9)
steps = 0

while numbers != sorted(numbers):
    print " ".join(map(str, numbers))
    flipcount = int(raw_input("Reverse how many? "))
    numbers[:flipcount] = reversed(numbers[:flipcount])
    steps += 1

print "Done! That took you %d steps." % steps
